controllers/etcdcluster_controller.go

```diff
@@ -79,6 +79,12 @@ func (r *EtcdClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// Validate in case a validating webhook has not been deployed
+	err := cluster.ValidateCreate()
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("invalid cluster: %w", err)
```

Comment:
Retrying won't make this resource well-formed. Instead of returning an error here, we should print a log message/set a condition/fire an event and `return nil` so we only resync once the resource has been updated.

Reply:
Ah yes, good point.